haystack/document_stores/memory.py

```diff
@@ -146,6 +172,25 @@ def write_documents(
                     )
                     continue
             self.indexes[index][document.id] = document
+            modified_documents += 1
+
+        if self.use_bm25 is True and modified_documents > 0:
+            self.update_bm25(index=index)
+
+    def update_bm25(self, index: Optional[str] = None):
+        """
+        Updates the BM25 sparse representation in the the document store.
+
+        :param index: Index name for which the BM25 representation is to be updated. If set to None, the default self.index is used.
+        """
+        index = index or self.index
+        tokenizer = re.compile(self.bm25_tokenization_regex).findall
+
+        logger.info("Updating BM25 representation...")
+        all_documents = self.get_all_documents(index=index)
+        tokenized_corpus = [tokenizer(doc.content.lower()) for doc in all_documents]
```

Comment:
If you want a progress bar to show, we can use `tqdm` here! Something like (untested):

```python
tokenized_corpus = [
    tokenizer(doc.content.lower()) 
    for doc in tqdm(self.get_all_documents(index=index), unit=" docs", desc="Updating BM25 representation...")
]
```